Makefile

```diff
@@ -374,6 +382,22 @@ unit-test: envtest
 	OPERATOR_NAMESPACE=$(OPERATOR_NAMESPACE) KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go test $(TEST_SRC) -v  -coverprofile cover.out
 CLEANFILES += cover.out
 
+.PHONY: extract-alert-rules
+extract-alert-rules: $(PROMETHEUS_CONFIG_YAML) 
+	./tests/scripts/extract_alerts.sh $(PROMETHEUS_CONFIG_YAML) $(GENERATED_ALERT_DIR)
+
+# Run prometheus-alert-unit-tests
+.PHONY: test-alerts
+test-alerts: extract-alert-rules
+	$(CONTAINER_RUNTIME) run --rm -t \
+	    -v "$(GENERATED_ALERT_DIR)":/prometheus/unit_tests:Z --entrypoint=/bin/sh \
+	    $(PROMETHEUS_IMAGE) -c 'cd unit_tests && promtool test rules *_unit_tests.yaml'
+
+#Check for alerts without unit-tests
+.PHONY: check-unit-tests
+check-unit-tests: extract-alert-rules
```

Comment:
Is it really a unit test? Makes confusion for me. Sounds like just independent prometheus tests

Reply:
> Maybe something more explicit like `check-prometheus-alert-unit-tests`

Why `unit`?

Reply:
@biswassri @StevenTobin this appears to still be an open question.

Reply:
A unit test in my view is a test against small pieces of code, in this case we are testing the alerts individually and hence is a set of unit tests for each these alert definitions.

But really I have no particular preference for keeping or removing unit. If there is confusion just remove it.

Reply:
@StevenTobin I think if we remove it here, we should ideally change the names of the alert test files too. So, I'm not sure. WDYT?

Reply:
I'm fine with it if it's only me who finds it it confusing.